gslb/ingestion/event_handlers.go

```diff
@@ -696,3 +696,186 @@ func AddHostRuleEventHandler(numWorkers uint32, c *GSLBMemberController) cache.R
+				"rejected ADD ingress because IP address/Hostname not found in status field")
+			continue
+		}
+		if !filter.ApplyFilter(filter.FilterArgs{
+			Obj:     ihm,
+			Cluster: c.name,
+		}) {
+			AddOrUpdateMultiClusterIngressStore(rejectedIngStore, ihm, c.name)
+			gslbutils.Logf("cluster: %s, ns: %s, ingress: %s, msg: %s, ing: %v\n", c.name, ihm.Namespace,
+				ihm.ObjName, "rejected ADD ingress key because it couldn't pass through the filter", ihm)
+			continue
+		}
+		AddOrUpdateMultiClusterIngressStore(acceptedIngStore, ihm, c.name)
+		if !fullsync {
+			publishKeyToGraphLayer(numWorkers, gslbutils.MCIType, c.name,
+				ihm.Namespace, ihm.ObjName, gslbutils.ObjectAdd, ihm.Hostname, c.workqueue)
+		}
+	}
+}
+
+func filterAndUpdateMultiClusterIngressMeta(oldIngMetaObjs, newIngMetaObjs []k8sobjects.MultiClusterIngressHostMeta, c *GSLBMemberController,
+	acceptedStore, rejectedStore *store.ClusterStore, numWorkers uint32) {
+
+	for _, mcihm := range oldIngMetaObjs {
+		// Check whether this exists in the new ingressHost list, if not, we need
+		// to delete this ingressHost object
+		newMCIhm, found := mcihm.IngressHostInList(newIngMetaObjs)
+		if !found {
+			// ingressHost doesn't exist anymore, delete this ingressHost object
+			_, isAccepted := acceptedStore.GetClusterNSObjectByName(c.name, mcihm.Namespace,
+				mcihm.ObjName)
+			DeleteFromMultiClusterIngressStore(acceptedStore, mcihm, c.name)
+			DeleteFromMultiClusterIngressStore(rejectedStore, mcihm, c.name)
+			// If part of accepted store, only then publish the delete key
+			if isAccepted {
+				publishKeyToGraphLayer(numWorkers, gslbutils.MCIType, c.name,
+					mcihm.Namespace, mcihm.ObjName, gslbutils.ObjectDelete, mcihm.Hostname, c.workqueue)
+			}
+			continue
+		}
+		// ingressHost exists, check if that got updated
+		if mcihm.GetIngressHostCksum() == newMCIhm.GetIngressHostCksum() {
+			// no changes, just continue
+			continue
+		}
+		// there are changes, need to send an update key, but first apply the filter
+		if !filter.ApplyFilter(filter.FilterArgs{
+			Obj:     newMCIhm,
+			Cluster: c.name,
+		}) {
+			// See if the ingressHost was already accepted, if yes, need to delete the key
+			fetchedObj, ok := acceptedStore.GetClusterNSObjectByName(c.name,
+				mcihm.Namespace, mcihm.ObjName)
+			if !ok {
+				// Nothing to be done, just add to the rejected ingress store
+				AddOrUpdateMultiClusterIngressStore(rejectedStore, newMCIhm, c.name)
+				continue
+			}
+			// Else, delete this ingressHost from accepted list and add the newIhm to the
+			// rejected store, and add a delete key for this ingressHost to the queue
+			AddOrUpdateMultiClusterIngressStore(rejectedStore, newMCIhm, c.name)
+			DeleteFromMultiClusterIngressStore(acceptedStore, newMCIhm, c.name)
+
+			fetchedIngHost := fetchedObj.(k8sobjects.IngressHostMeta)
+			// Add a DELETE key for this ingHost
+			publishKeyToGraphLayer(numWorkers, gslbutils.IngressType, fetchedIngHost.Cluster,
```

Comment:
Here the type has to be `gslbutils.MCIType` instead of `gslbutils.IngressType`

Reply:
I have corrected the object type passed to the graph layer.